server/sockets/list.js

```diff
@@ -477,6 +480,43 @@ const changeListType = (socket, dashboardClients, cohortClients, listClients) =>
       });
   });
 
+const removeListsOnArchiveCohort = (socket, dashboardClients) =>
+  socket.on(CohortActionTypes.ARCHIVE_SUCCESS, data => {
+    const { cohortId } = data;
+
+    List.find({
```

Comment:
I can see the pattern being replicated many times. What about a helper?

```javascript
resolveMultiple => Model => selector => callback =>
  Model
    .find(selector)
    .lean()
    .exec()
    .then(docs => {
      if (docs) {
        const ids= docs.map(({ _id })=> _id.toString());
        callback(docs, ids);
      }
    })

resolveSingle => Model => selector => callback =>  
  Model
    .find(selector)
    .lean()
    .exec()
    .then(docs => {
      if (doc) {
        callback(doc, doc._id.toString());
      }
    })

resolveMultiple(List)({ cohortId })((docs, ids) => {
  const listsByViewers = getListsByViewers(docs);

  ids.forEach(listId => { ... });
});
```

Reply:
Great idea, but I think we should refactor emitting sockets and move emitting events from frontend actions to the backend controllers (we discussed it on Wednesday about the security of our current approach) and in controllers, the sockets will have access to all needed data so additional queries to DB would be unnecessary (as we   do it now). But I save your proposal for future purposes :-)